Docs work (#194)
MystenLabs/suins-contracts: documentation/pages/move-registry/design.mdx

# Design of the Move Registry

> Before diving into specifics, we first need to analyze the design. The Move Registry consists of two independent components:

Reviewer: ```suggestion
Before diving into specifics, let's understand its design. The Move Registry consists of two independent components:
```

Sui indexers can index the metadata as it cannot be stored within another object.

![Package Info Objects](/images/mvr/mvr-package-info.svg)

See [Managing Package Metadata](./managing-package-info.mdx) for examples of working with `PackageInfo` objects. 

## MVR registration

MVR registration is separate from the `PackageInfo` registration process. This separation is necessary because the `PackageInfo` object acts as both the source of truth for package metadata and a proof of package ownership, whereas MVR registers applications that MVR tooling can resolve.

To register a package:

1. Using your SuiNS name, register an application.
2. Based on the network, associate the application name with a `PackageInfo` object:
- **Mainnet:** Associate the application name with a complete `PackageInfo` object, which provides the full metadata and ownership proof.
- **Other networks:** Associate the application name with the `PackageInfo` object ID, acting as a pointer without ensuring a full mapping.

![Move Registry](/images/mvr/mvr-registry.svg)

<Callout type="info" emoji="ℹ️">

> The single source of truth for MVR exists on Mainnet.

Reviewer 2: Could we say more about what "single source of truth" means? Such as, "The single source of truth for MVR exists on Mainnet. This means that..."
